app/__test__/product/product.schema.test.ts

```diff
@@ -0,0 +1,253 @@
+});
+
+it('should throw an error when the product name has length greater than 50', () => {
+  const invalidInput = {
+    name: 'Product Name with a very long length that exceeds the maximum limit of 50 characters',
+    description: 'Product Description',
+    price: 10,
+    ProductCategory: [
+      {
+        category_id: 1
+      }
+    ],
+    Attribute: [
+      {
+        size: 'M',
+        color: '#FFFFFF',
+        quantity: 5,
+        image: 'https://example.com/image.jpg'
+      }
+    ]
+  };
+
+  const result = createProductSchema.safeParse(invalidInput);
+
+  expect(result.success).toBe(false);
+  expect(result.error.issues[0].message).toBe('Maximum characters are 50');
+});
+
+it('should throw an error when the product description has length less than 20', () => {
+  const invalidInput = {
+    name: 'Product Name',
+    description: 'Product',
+    price: 10,
+    ProductCategory: [
+      {
+        category_id: 1
+      }
+    ],
+    Attribute: [
+      {
+        size: 'M',
+        color: '#FFFFFF',
+        quantity: 5,
+        image: 'https://example.com/image.jpg'
+      }
+    ]
+  };
+
+  const result = createProductSchema.safeParse(invalidInput);
+
+  expect(result.success).toBe(false);
+  expect(result.error.issues[0].message).toBe('Minium of 20 characters required');
+});
+
+it('should throw an error when the product description has length greater than 200', () => {
+  const invalidInput = {
+    name: 'Product Name',
+    description: 'Product Description with a very long length that exceeds the maximum limit of 200 characters Product Description with a very long length that exceeds the maximum limit of 200 characters Product Description with a very long length that exceeds the maximum limit of 200 characters',
+    price: 10,
+    ProductCategory: [
+      {
+        category_id: 1
+      }
+    ],
+    Attribute: [
+      {
+        size: 'M',
+        color: '#FFFFFF',
+        quantity: 5,
+        image: 'https://example.com/image.jpg'
+      }
+    ]
+  };
+
+  const result = createProductSchema.safeParse(invalidInput);
+
+  expect(result.success).toBe(false);
+  expect(result.error.issues[0].message).toBe('Maximun characters are 200');
+});
+
+it('should throw an error when the attribute has length less than 10', () => {
+  const invalidInput = {
+    name: 'Product Name',
+    description: 'Product Description',
+    price: 10,
+    ProductCategory: [
+      {
+        category_id: 1
+      }
+    ],
+    Attribute: [
+      {
+        size: 'M',
+        color: '#FFFFFF',
+        quantity: 5,
+        image: 'https://example.com/image.jpg'
+      },
+      {
+        size: 'S',
+        color: '#000000',
+        quantity: 3,
+        image: 'https://example.com/image.jpg'
+      }
+    ]
+  };
+
+  const result = createProductSchema.safeParse(invalidInput);
+
+  expect(result.success).toBe(false);
+  expect(result.error.issues[0].message).toBe('Minium of 20 characters required');
```

Comment:
hmm do you have a collection of issues? why you are choosing the first one? (issues[0])

Reply:
It is an array of issues so, in this case that only have an error by input it is showed in the issue[0]